src/server/game/Handlers/CalendarHandler.cpp

```diff
@@ -318,6 +327,14 @@ void WorldSession::HandleCalendarUpdateEvent(WorldPacket& recvData)
     recvData.ReadPackedTime(timeZoneTime);
     recvData >> flags;
 
+    // prevent attacks with non-utf8 chars -> with multiple packets it will hang up the db due to errors.
+    if (!utf8::is_valid(title.begin(), title.end()) || !utf8::is_valid(description.begin(), description.end()))
+    {
+        sLog->outString("CalendarHandler: Player with guid %u attempt to update an event with invalid name (packet modification)", guid);
+        recvData.rfinish();
+        return;
+    }
```

Comment:
good job! what about wrapping this logic into a method so we can reuse it, instead of duplicating it?

as I can see the only difference is in the `outString`, one says `updated` and the other one `create`.

for example `checkUpdated(type)` where `type` can either be `create` or `update`.

Reply:
@FrancescoBorzi https://github.com/azerothcore/azerothcore-wotlk/pull/2721/files#diff-d66ed2e0118814a930a18391f6fb5ec5R307-R331